File: fontes/backend/java/fenix/src/main/java/com/t2ti/fenix/model/folha/FolhaParametro.java
/*******************************************************************************
Title: T2Ti ERP 3.0                                                                
Description: Model relacionado à tabela [FOLHA_PARAMETRO] 
                                                                                
The MIT License                                                                 
                                                                                
Copyright: Copyright (C) 2021 T2Ti.COM                                          
                                                                                
Permission is hereby granted, free of charge, to any person                     
obtaining a copy of this software and associated documentation                  
files (the "Software"), to deal in the Software without                         
restriction, including without limitation the rights to use,                    
copy, modify, merge, publish, distribute, sublicense, and/or sell               
copies of the Software, and to permit persons to whom the                       
Software is furnished to do so, subject to the following                        
conditions:                                                                     
                                                                                
The above copyright notice and this permission notice shall be                  
included in all copies or substantial portions of the Software.                 
                                                                                
THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,                 
EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES                 
OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND                        
NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT                     
HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,                    
WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING                    
FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR                   
OTHER DEALINGS IN THE SOFTWARE.                                                 
                                                                                
       The author may be contacted at:                                          
           t2ti.com@gmail.com                                                   
                                                                                
@author Albert Eije (alberteije@gmail.com)                    
@version 1.0.0
*******************************************************************************/
package com.t2ti.fenix.model.folha;

import java.io.Serializable;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.NamedQuery;
import javax.persistence.Table;
import java.math.BigDecimal;

@Entity
@Table(name="FOLHA_PARAMETRO")
@NamedQuery(name="FolhaParametro.findAll", query="SELECT t FROM FolhaParametro t")
public class FolhaParametro implements Serializable {
	private static final long serialVersionUID = 1L;

    @Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Integer id;

    @Column(name="COMPETENCIA")
	private String competencia;

    @Column(name="CONTRIBUI_PIS")
	private String contribuiPis;

    @Column(name="ALIQUOTA_PIS")
	private BigDecimal aliquotaPis;

    @Column(name="DISCRIMINAR_DSR")
	private String discriminarDsr;

    @Column(name="DIA_PAGAMENTO")
	private String diaPagamento;

    @Column(name="CALCULO_PROPORCIONALIDADE")
	private String calculoProporcionalidade;

    @Column(name="DESCONTAR_FALTAS_13")
	private String descontarFaltas13;

    @Column(name="PAGAR_ADICIONAIS_13")
	private String pagarAdicionais13;

    @Column(name="PAGAR_ESTAGIARIOS_13")
	private String pagarEstagiarios13;

    @Column(name="MES_ADIANTAMENTO_13")
	private String mesAdiantamento13;

    @Column(name="PERCENTUAL_ADIANTAM_13")
	private BigDecimal percentualAdiantam13;

    @Column(name="FERIAS_DESCONTAR_FALTAS")
	private String feriasDescontarFaltas;

    @Column(name="FERIAS_PAGAR_ADICIONAIS")
	private String feriasPagarAdicionais;

    @Column(name="FERIAS_ADIANTAR_13")
	private String feriasAdiantar13;

    @Column(name="FERIAS_PAGAR_ESTAGIARIOS")
	private String feriasPagarEstagiarios;

    @Column(name="FERIAS_CALC_JUSTA_CAUSA")
	private String feriasCalcJustaCausa;

    @Column(name="FERIAS_MOVIMENTO_MENSAL")
	private String feriasMovimentoMensal;

	public FolhaParametro() {
		//construtor padrão
	}

    public Integer getId() {
		return this.id;
	}

	public void setId(Integer id) {
		this.id = id;
	}

    public String getCompetencia() {
		return this.competencia;
	}

	public void setCompetencia(String competencia) {
		this.competencia = competencia;
	}

    public String getContribuiPis() {
		return this.contribuiPis;
	}

	public void setContribuiPis(String contribuiPis) {
		this.contribuiPis = contribuiPis;
	}

    public BigDecimal getAliquotaPis() {
		return this.aliquotaPis;
	}

	public void setAliquotaPis(BigDecimal aliquotaPis) {
		this.aliquotaPis = aliquotaPis;
	}

    public String getDiscriminarDsr() {
		return this.discriminarDsr;
	}

	public void setDiscriminarDsr(String discriminarDsr) {
		this.discriminarDsr = discriminarDsr;
	}

    public String getDiaPagamento() {
		return this.diaPagamento;
	}

	public void setDiaPagamento(String diaPagamento) {
		this.diaPagamento = diaPagamento;
	}

    public String getCalculoProporcionalidade() {
		return this.calculoProporcionalidade;
	}

	public void setCalculoProporcionalidade(String calculoProporcionalidade) {
		this.calculoProporcionalidade = calculoProporcionalidade;
	}

    public String getDescontarFaltas13() {
		return this.descontarFaltas13;
	}

	public void setDescontarFaltas13(String descontarFaltas13) {
		this.descontarFaltas13 = descontarFaltas13;
	}

    public String getPagarAdicionais13() {
		return this.pagarAdicionais13;
	}

	public void setPagarAdicionais13(String pagarAdicionais13) {
		this.pagarAdicionais13 = pagarAdicionais13;
	}

    public String getPagarEstagiarios13() {
		return this.pagarEstagiarios13;
	}

	public void setPagarEstagiarios13(String pagarEstagiarios13) {
		this.pagarEstagiarios13 = pagarEstagiarios13;
	}

    public String getMesAdiantamento13() {
		return this.mesAdiantamento13;
	}

	public void setMesAdiantamento13(String mesAdiantamento13) {
		this.mesAdiantamento13 = mesAdiantamento13;
	}

    public BigDecimal getPercentualAdiantam13() {
		return this.percentualAdiantam13;
	}

	public void setPercentualAdiantam13(BigDecimal percentualAdiantam13) {
		this.percentualAdiantam13 = percentualAdiantam13;
	}

    public String getFeriasDescontarFaltas() {
		return this.feriasDescontarFaltas;
	}

	public void setFeriasDescontarFaltas(String feriasDescontarFaltas) {
		this.feriasDescontarFaltas = feriasDescontarFaltas;
	}

    public String getFeriasPagarAdicionais() {
		return this.feriasPagarAdicionais;
	}

	public void setFeriasPagarAdicionais(String feriasPagarAdicionais) {
		this.feriasPagarAdicionais = feriasPagarAdicionais;
	}

    public String getFeriasAdiantar13() {
		return this.feriasAdiantar13;
	}

	public void setFeriasAdiantar13(String feriasAdiantar13) {
		this.feriasAdiantar13 = feriasAdiantar13;
	}

    public String getFeriasPagarEstagiarios() {
		return this.feriasPagarEstagiarios;
	}

	public void setFeriasPagarEstagiarios(String feriasPagarEstagiarios) {
		this.feriasPagarEstagiarios = feriasPagarEstagiarios;
	}

    public String getFeriasCalcJustaCausa() {
		return this.feriasCalcJustaCausa;
	}

	public void setFeriasCalcJustaCausa(String feriasCalcJustaCausa) {
		this.feriasCalcJustaCausa = feriasCalcJustaCausa;
	}

    public String getFeriasMovimentoMensal() {
		return this.feriasMovimentoMensal;
	}

	public void setFeriasMovimentoMensal(String feriasMovimentoMensal) {
		this.feriasMovimentoMensal = feriasMovimentoMensal;
	}

		
}

File: fontes/backend/delphi/dmvc/Model/Folha/FolhaParametro.pas
{*******************************************************************************
Title: T2Ti ERP 3.0                                                                
Description: Model relacionado à tabela [FOLHA_PARAMETRO] 
                                                                                
The MIT License                                                                 
                                                                                
Copyright: Copyright (C) 2021 T2Ti.COM                                          
                                                                                
Permission is hereby granted, free of charge, to any person                     
obtaining a copy of this software and associated documentation                  
files (the "Software"), to deal in the Software without                         
restriction, including without limitation the rights to use,                    
copy, modify, merge, publish, distribute, sublicense, and/or sell               
copies of the Software, and to permit persons to whom the                       
Software is furnished to do so, subject to the following                        
conditions:                                                                     
                                                                                
The above copyright notice and this permission notice shall be                  
included in all copies or substantial portions of the Software.                 
                                                                                
THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,                 
EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES                 
OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND                        
NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT                     
HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,                    
WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING                    
FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR                   
OTHER DEALINGS IN THE SOFTWARE.                                                 
                                                                                
       The author may be contacted at:                                          
           t2ti.com@gmail.com                                                   
                                                                                
@author Albert Eije (alberteije@gmail.com)                    
@version 1.0.0
*******************************************************************************}
unit FolhaParametro;

interface

uses
  MVCFramework.Serializer.Commons, ModelBase;

type

  [MVCNameCase(ncCamelCase)]
  TFolhaParametro = class(TModelBase)
  private
    FId: Integer;
    FCompetencia: string;
    FContribuiPis: string;
    FAliquotaPis: Extended;
    FDiscriminarDsr: string;
    FDiaPagamento: string;
    FCalculoProporcionalidade: string;
    FDescontarFaltas13: string;
    FPagarAdicionais13: string;
    FPagarEstagiarios13: string;
    FMesAdiantamento13: string;
    FPercentualAdiantam13: Extended;
    FFeriasDescontarFaltas: string;
    FFeriasPagarAdicionais: string;
    FFeriasAdiantar13: string;
    FFeriasPagarEstagiarios: string;
    FFeriasCalcJustaCausa: string;
    FFeriasMovimentoMensal: string;
      
  public
    //procedure ValidarInsercao; override;
    //procedure ValidarAlteracao; override;
    //procedure ValidarExclusao; override;

    [MVCColumnAttribute('ID', True)]
		[MVCNameAsAttribute('id')]
		property Id: Integer read FId write FId;
    [MVCColumnAttribute('COMPETENCIA')]
		[MVCNameAsAttribute('competencia')]
		property Competencia: string read FCompetencia write FCompetencia;
    [MVCColumnAttribute('CONTRIBUI_PIS')]
		[MVCNameAsAttribute('contribuiPis')]
		property ContribuiPis: string read FContribuiPis write FContribuiPis;
    [MVCColumnAttribute('ALIQUOTA_PIS')]
		[MVCNameAsAttribute('aliquotaPis')]
		property AliquotaPis: Extended read FAliquotaPis write FAliquotaPis;
    [MVCColumnAttribute('DISCRIMINAR_DSR')]
		[MVCNameAsAttribute('discriminarDsr')]
		property DiscriminarDsr: string read FDiscriminarDsr write FDiscriminarDsr;
    [MVCColumnAttribute('DIA_PAGAMENTO')]
		[MVCNameAsAttribute('diaPagamento')]
		property DiaPagamento: string read FDiaPagamento write FDiaPagamento;
    [MVCColumnAttribute('CALCULO_PROPORCIONALIDADE')]
		[MVCNameAsAttribute('calculoProporcionalidade')]
		property CalculoProporcionalidade: string read FCalculoProporcionalidade write FCalculoProporcionalidade;
    [MVCColumnAttribute('DESCONTAR_FALTAS_13')]
		[MVCNameAsAttribute('descontarFaltas13')]
		property DescontarFaltas13: string read FDescontarFaltas13 write FDescontarFaltas13;
    [MVCColumnAttribute('PAGAR_ADICIONAIS_13')]
		[MVCNameAsAttribute('pagarAdicionais13')]
		property PagarAdicionais13: string read FPagarAdicionais13 write FPagarAdicionais13;
    [MVCColumnAttribute('PAGAR_ESTAGIARIOS_13')]
		[MVCNameAsAttribute('pagarEstagiarios13')]
		property PagarEstagiarios13: string read FPagarEstagiarios13 write FPagarEstagiarios13;
    [MVCColumnAttribute('MES_ADIANTAMENTO_13')]
		[MVCNameAsAttribute('mesAdiantamento13')]
		property MesAdiantamento13: string read FMesAdiantamento13 write FMesAdiantamento13;
    [MVCColumnAttribute('PERCENTUAL_ADIANTAM_13')]
		[MVCNameAsAttribute('percentualAdiantam13')]
		property PercentualAdiantam13: Extended read FPercentualAdiantam13 write FPercentualAdiantam13;
    [MVCColumnAttribute('FERIAS_DESCONTAR_FALTAS')]
		[MVCNameAsAttribute('feriasDescontarFaltas')]
		property FeriasDescontarFaltas: string read FFeriasDescontarFaltas write FFeriasDescontarFaltas;
    [MVCColumnAttribute('FERIAS_PAGAR_ADICIONAIS')]
		[MVCNameAsAttribute('feriasPagarAdicionais')]
		property FeriasPagarAdicionais: string read FFeriasPagarAdicionais write FFeriasPagarAdicionais;
    [MVCColumnAttribute('FERIAS_ADIANTAR_13')]
		[MVCNameAsAttribute('feriasAdiantar13')]
		property FeriasAdiantar13: string read FFeriasAdiantar13 write FFeriasAdiantar13;
    [MVCColumnAttribute('FERIAS_PAGAR_ESTAGIARIOS')]
		[MVCNameAsAttribute('feriasPagarEstagiarios')]
		property FeriasPagarEstagiarios: string read FFeriasPagarEstagiarios write FFeriasPagarEstagiarios;
    [MVCColumnAttribute('FERIAS_CALC_JUSTA_CAUSA')]
		[MVCNameAsAttribute('feriasCalcJustaCausa')]
		property FeriasCalcJustaCausa: string read FFeriasCalcJustaCausa write FFeriasCalcJustaCausa;
    [MVCColumnAttribute('FERIAS_MOVIMENTO_MENSAL')]
		[MVCNameAsAttribute('feriasMovimentoMensal')]
		property FeriasMovimentoMensal: string read FFeriasMovimentoMensal write FFeriasMovimentoMensal;
      
  end;

implementation

{ TFolhaParametro }



end.